server/core/src/catalog.cpp

```diff
@@ -0,0 +1,93 @@
+#include <fstream>
+#include <functional>
+#include <stdexcept>
+
+#include "irods_configuration_keywords.hpp"
+#include "irods_get_full_path_for_config_file.hpp"
+#include "irods_logger.hpp"
+#include "irods_stacktrace.hpp"
+#include "objDesc.hpp"
+#include "rodsConnect.h"
+
+#include "catalog.hpp"
+
+#include "json.hpp"
+#include "fmt/format.h"
+#include "nanodbc/nanodbc.h"
+
+namespace irods::experimental::catalog {
```

Comment:
might consider some ::trace calls in here if someone is debugging...

Reply:
I will add the DSN string when the connection attempt is made. Otherwise, I think the error messages sent back from nanodbc and our own exceptions should be sufficient without leaking sensitive information in the logs.